### ISSUE TITLE
Can't login into ombi using emby user

### ISSUE THREAD

**Issue by gpulido** [tidusjar/Ombi#2546]
#### Ombi build Version:

V 3.0.3795

#### Update Branch:

master

#### Media Sever:

Emby

#### Media Server Version:

3.5.3.0

#### Operating System:

Using docker container: linuxserver/ombi:latest

#### Ombi Applicable Logs (from `/logs/` directory or the Admin page):

```
2018-09-28 13:21:52.375 +02:00 [Error] StatusCode: InternalServerError, Reason: Internal Server Error, RequestUri: http://emby.myhost.com/emby/users/authenticatebyname

```

#### Problem Description:

Can not use emby users on ombi to log. The rest of emby integration through API key seems to work.
On the emby log part I'm receiving the following:
```
2018-09-28 13:21:52.373 Info HttpServer: HTTP GET http://emby.myhost.com:8096/emby/users/authenticatebyname. UserAgent: Ombi
2018-09-28 13:21:52.374 Error HttpServer: Error processing request
	*** Error Report ***
	Version: 3.5.3.0
	Command line: /system/EmbyServer.dll -programdata /config -ffmpeg /bin/ffmpeg -ffprobe /bin/ffprobe -restartexitcode 3
	Operating system: Unix 4.15.0.34
	64-Bit OS: True
	64-Bit Process: True
	User Interactive: True
	Processor count: 8
	Program data path: /config
	Application directory: /system
	System.FormatException: Input string was not in a correct format.
	   at ServiceStack.Text.StringSegmentExtensions.ParseGeneralStyleGuid(StringSegment value, Int32& len)
	   at ServiceStack.Text.StringSegmentExtensions.ParseGuid(StringSegment value)
	   at ServiceStack.Text.Common.DeserializeBuiltin`1.<>c.<GetParseStringSegmentFn>b__7_14(StringSegment value)
	   at ServiceStack.Text.Jsv.JsvReader.<>c__DisplayClass2_0.<GetParseFn>b__0(String v)
	   at Emby.Server.Implementations.Services.StringMapTypeDeserializer.PopulateFromMap(Object instance, IDictionary`2 keyValuePairs)
	   at Emby.Server.Implementations.Services.RestPath.CreateRequest(String pathInfo, Dictionary`2 queryStringAndFormData, Object fromInstance)
	   at Emby.Server.Implementations.Services.ServiceHandler.CreateRequest(IRequest httpReq, RestPath restPath, Dictionary`2 requestParams, Object requestDto)
	   at Emby.Server.Implementations.Services.ServiceHandler.CreateRequest(HttpListenerHost host, IRequest httpReq, RestPath restPath, ILogger logger)
	   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost appHost, IRequest httpReq, IResponse httpRes, ILogger logger, String operationName, CancellationToken cancellationToken)
	   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
	System.FormatException
	   at ServiceStack.Text.StringSegmentExtensions.ParseGeneralStyleGuid(StringSegment value, Int32& len)
	   at ServiceStack.Text.StringSegmentExtensions.ParseGuid(StringSegment value)
	   at ServiceStack.Text.Common.DeserializeBuiltin`1.<>c.<GetParseStringSegmentFn>b__7_14(StringSegment value)
	   at ServiceStack.Text.Jsv.JsvReader.<>c__DisplayClass2_0.<GetParseFn>b__0(String v)
	   at Emby.Server.Implementations.Services.StringMapTypeDeserializer.PopulateFromMap(Object instance, IDictionary`2 keyValuePairs)
	   at Emby.Server.Implementations.Services.RestPath.CreateRequest(String pathInfo, Dictionary`2 queryStringAndFormData, Object fromInstance)
	   at Emby.Server.Implementations.Services.ServiceHandler.CreateRequest(IRequest httpReq, RestPath restPath, Dictionary`2 requestParams, Object requestDto)
	   at Emby.Server.Implementations.Services.ServiceHandler.CreateRequest(HttpListenerHost host, IRequest httpReq, RestPath restPath, ILogger logger)
	   at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost appHost, IRequest httpReq, IResponse httpRes, ILogger logger, String operationName, CancellationToken cancellationToken)
	   at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)
	
2018-09-28 13:21:52.374 Info HttpServer: HTTP Response 500 to 10.255.0.2. Time: 1ms. http://emby.myhost.com:8096/emby/users/authenticatebyname 
```

#### Reproduction Steps:

Just create an user on emby, import in ombi and try to log in ombi using it.
Also After reviewing the EmbyApi.cs file it seems that it is not updated with the emby api, that only needs the username and pw.

https://github.com/MediaBrowser/Emby/wiki/User-Authentication#authenticating-a-user

**Comment by ddurdle** [comment]
can we get this fixed?

**Comment by tidusjar** [comment]
I cannot seem to recreate this with a Emby Connect or a Local emby user.

Everything seems fine to me.

**Comment by tidusjar** [comment]
Is the local emby user in the ombi user management page

**Comment by gpulido** [comment]
Yes, I have imported it using the import users from emby functionality and it showns on the user management page as "Emby User"

**Comment by tidusjar** [comment]
Are you attempting to log in with the password or pin

**Comment by tidusjar** [comment]
I cannot recreate that issue,

Seems to work perfectly fine. One thing I can think of is that we authenticate with the local server, so the emby server that is in your settings is 100% the server the user is located on?

**Comment by gpulido** [comment]
Ok, after your last comment it make me think if it was a DNS / traefik redirection issue.
So Instead of configure ombi to use emby.mydomain.com:80 (where traefik is exposing it to the outside world) I changed it to "docker_host_local_network_ip:8096" (I have mapped the emby 8096 port from the container to the host, and it worked!!.
What is strange is that using emby.mydomain.com ombi was able to connect to emby to retrieve users and communicate data, so it seems that maybe ombi is not using the right ip / name / port to access the userauthentication api on the emby server. Reviewing the logs, it seems that even if I set ombi to use the port 80 to access emby it was trying to use the port 8096 for userauth.
Could be that the problem?
